lambdaclass/elixir_riak_core
# Serve CSV from riak

## Context:
- The idea is to provide an example of use case for Riak that is not a simple key-value. The code here would be used to set up an example like explained in [here](https://hackmd.io/8BFqFT7IRq-8ZHnaHhtApg?both). The goal is to show case how to use Riak as a way to distribute several files (or one big file) in memory.
## Changes
### Added RiaxCSV module for tutorial:
  - This change is to provide a function to store a CSV as index -> row pairs distributed among running nodes.
### Update VNode for tutorial:
  - I've added a new handle_command clause to store the file without logging.
### CSV Visualization
- I've added a Liveview and JSON views for the rows of the csv file, where it is shown both the csv's row info and the node that is storing it.

## How to test it:
If you really want to test everything, you can follow this steps:

1.  Download the CSV file from [here](https://www.kaggle.com/datasets/gauravduttakiit/bitcoin-tweets-16m-tweets-with-sentiment-tagged?resource=download%29).
 This is just a CSV of tweets about bitcoin.
2. Run mix deps.get
3. Set up the 4 nodes on different terminals with `make node1_limited`, `make node2_limited`, `make node3_limited` and `make phoenix_limited`. 
4. Join the nodes with `:riak_core.join(node@host)`.
5. Run `Riax.CSV.distribute_csv(path)`, with the argument being a string with the path to the CSV file.
    **Be ware this will probably consume a lot of memory.**
6. Now you should be able to visit: http://localhost:4003/api/num or http://localhost:4003/csv/num, num > 0.
